docs/types/mapping-types.rst

```diff
@@ -1,44 +1,24 @@
-``ValueType`` can be any type, including mappings, arrays and structs.
-
-You can think of mappings as `hash tables <https://en.wikipedia.org/wiki/Hash_table>`_, which are virtually initialised
-such that every possible key exists and is mapped to a value whose
-byte-representation is all zeros, a type's :ref:`default value <default-value>`.
-The similarity ends there, the key data is not stored in a
-mapping, only its ``keccak256`` hash is used to look up the value.
-
-Because of this, mappings do not have a length or a concept of a key or
-value being set, and therefore cannot be erased without extra information
-regarding the assigned keys (see :ref:`clearing-mappings`).
-
-Mappings can only have a data location of ``storage`` and thus
-are allowed for state variables, as storage reference types
-in functions, or as parameters for library functions.
-They cannot be used as parameters or return parameters
-of contract functions that are publicly visible.
-These restrictions are also true for arrays and structs that contain mappings.
-
-You can mark state variables of mapping type as ``public`` and Solidity creates a
-:ref:`getter <visibility-and-getters>` for you. The ``KeyType`` becomes a parameter for the getter.
-If ``ValueType`` is a value type or a struct, the getter returns ``ValueType``.
-If ``ValueType`` is an array or a mapping, the getter has one parameter for
-each ``KeyType``, recursively.
-
-In the example below, the ``MappingExample`` contract defines a public ``balances``
-mapping, with the key type an ``address``, and a value type a ``uint``, mapping
-an Ethereum address to an unsigned integer value. As ``uint`` is a value type, the getter
-returns a value that matches the type, which you can see in the ``MappingUser``
-contract that returns the value at the specified address.
+Eşleme türleri ``mapping(KeyType => ValueType)`` sözdizimi yapısını kullanır ve eşleme türünün değişlenleri, ``mapping(KeyType => ValueType) VariableName``sözdizimi kullanılarak bildirilir.
+
+``KeyType``, herhangi bir yerleşik değer türü, ``bytes``, ``string``, herhangi bir sözleşme ya da numaralandırma türü olabilir. Eşlemeler, yapılar veya dizi türleri gibi diğer kullanıcı tanımlı veya karmaşık türlere izin verilmez. ``ValueType``, eşlemeleri, dizileri ve yapıları içeren herhangi bir tür olabilir.  
+
+
+Eşlemeleri, olası her anahtarın var olduğu ve bir türün :ref:`varsayılan değeri <default-value>` olan bayt temsilinin tamamı sıfır olan bir değere eşlendiği şekilde sanal olarak başlatılan `karma tablolar <https://en.wikipedia.org/wiki/Hash_table>`_ olarak düşünebilirsiniz. Benzerlik burada sona eriyor, anahtar veriler bir eşlemede saklanmıyor, değeri aramak için yalnızca ``keccak256`` karma değeri kullanılıyor.
+
+Bu nedenle, eşlemelerin bir uzunluğu veya ayarlanan bir anahtar veya değer kavramı yoktur ve bu nedenle atanan anahtarlarla ilgili ek bilgi olmadan silinemezler (bkz. :ref:`eşlemeleri-silme`).
```

Comment:
```suggestion
Bu nedenle, eşlemelerin bir uzunluğu veya ayarlanan bir anahtar veya değer kavramı yoktur ve bu nedenle atanan anahtarlarla ilgili ek bilgi olmadan silinemezler (bkz. :ref:`eşlemeleri-silme <clearing-mappings>`).
```

ref'in referans gösterdiği identifier'ı değiştirmeyelim ama hyperlink'in metnini değiştirebiliriz yukarıda yaptığımız gibi